test-case/check-kmod-load-unload.sh

```diff
@@ -50,6 +50,16 @@ do
     ## - 1: remove module section
     func_lib_setup_kernel_last_line
 
+    # 1. check dsp status before removal, only remove module when dsp is suspended.
+    # 2. it will take some time (about 10s) for dsp to become suspended from active
+    # after module insertion on some platforms, here we check dsp status, and only continue
+    # next removal when dsp is suspended.
+    for i in $(seq 1 15)
+    do
+        [[ $(cat /sys/devices/pci0000:00/0000:00:1f.3/power/runtime_status) == "suspended" ]] && break
```

Comment:
here you device id is hard code, i think you can catch it from `sof-dump-status.py`   
https://github.com/thesofproject/sof-test/blob/master/tools/sof-dump-status.py#L168
```
        for run_status in self.sys_power['run_status']:
            exit_code, output=subprocess.getstatusoutput("cat /sys/bus/pci/devices/0000:%s/power/runtime_status" % (run_status['pci_id']))
            if exit_code != 0:
                continue
            run_status['status'] = output
```
so here you can add option for it

Reply:
@Bin-QA you are right, the pci id is not unique on different platforms. has added an option in sof-dump-status.py, and modified module reload case to call sof-dump-status.py to get current dsp status.